File: Software Development Module/Revision/Exam2018/SentanceEncoderApp.java
/*
SentanceEncoderApp.java
Billy Cussen
02/09/2020
*/

import java.util.InputMismatchException;
import java.util.Scanner;

public class SentanceEncoderApp{

    public static void main(String [] args){

        //Variables
        String inputSentance = new String();
        boolean inputCheck, spaceCheck, dotCheck;

        //Objects
        Scanner keyboard;
        SentanceEncoder mySentanceEncoder;

        do{
            inputCheck = true;
            spaceCheck = false;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Enter the Sentance to be Encoded: ");
                inputSentance = keyboard.nextLine();

                for(int i = 0; i < inputSentance.length(); i++){
                    if(inputSentance.charAt(i)==' '){
                        spaceCheck = true;
                        break;
                    }
                }

                if(!(spaceCheck && inputSentance.charAt(inputSentance.length()-1)=='.' && inputSentance.length() >= 8)){
                    throw new InputMismatchException("You need to enter a Sentance with a Full Stop at the end and a Character length of 8 or Greater!");
                }

            }catch(InputMismatchException e){
                System.out.println(e.getMessage());
                inputCheck = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);

        //Process & Output
        mySentanceEncoder = new SentanceEncoder();
        long startTime = System.currentTimeMillis();
        mySentanceEncoder.getEncodedSentance(inputSentance);
        long endTime = System.currentTimeMillis();
        System.out.println("This function took: "+(endTime-startTime)+" milliseconds");

        mySentanceEncoder = null;

    }

}

File: Python/Software Development Module - Python/Revision/Exam2018/SentanceEncoder/SentanceEncoderApp.py
"""
SentanceEncoderApp.py
Billy Cussen
02/11/2020
"""

#Object
import re
import gc
import SentanceEncoder
from SentanceEncoder import mySentanceEncoder

#Variable
repeat = "y"
regex = re.compile('[@_!#$%^&*()<>?/\|}{~:]') 

while repeat == "y":
    spaceCheck = False
    digitCheck = False
    inputCheck = False

    while inputCheck == False:
        try:
            sentance = str(input("Please enter the sentance you would like encoded: "))
            for c in sentance:
                if c == ' ':
                    spaceCheck = True
                if c.isdigit():
                    digitCheck = True
            if spaceCheck == False or digitCheck or sentance[-1]!='.' or regex.search(sentance)!=None:
                raise Exception
        except Exception:
            print("You need to enter a Sentance with only Letters, Spaces and Full Stops")
        else:
            inputCheck = True

    #Process & Output
    mySentanceEncoderApp = mySentanceEncoder(sentance)
    mySentanceEncoderApp.getEncodedSentance()
    mySentanceEncoderApp = None
    gc.collect()

    repeat = ""

    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"y\" to retry, enter \"n\" to exit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat !="n":
            print ("You need to enter \"y\" or \"n\".")
        
print("Thanks for using my Sentance Encoder App!")